its/ruling/src/test/java/org/sonar/java/it/JavaRulingTest.java

```diff
@@ -92,17 +93,87 @@ public void test() throws Exception {
+    File litsDifferencesFile = FileLocation.of("target/differences").getFile();
+    File pomFile = FileLocation.of("../sources/commons-beanutils/pom.xml").getFile();
+    MavenBuild mavenBuild = MavenBuild.create().setPom(pomFile).setCleanPackageSonarGoals().addArgument("-DskipTests")
+        .setProfile("rules")
+        .setProperty("sonar.cpd.skip", "true")
+        .setProperty("sonar.skipPackageDesign", "true")
+        .setProperty("sonar.analysis.mode", "preview")
+        .setProperty("sonar.issuesReport.html.enable", "true")
+        .setProperty("dump.old", FileLocation.of("src/test/resources/commons-beanutils").getFile().getAbsolutePath())
+        .setProperty("dump.new", FileLocation.of("target/actual/commons-beanutils").getFile().getAbsolutePath())
+        .setProperty("lits.differences", litsDifferencesFile.getAbsolutePath());
+    orchestrator.executeBuild(mavenBuild);
+    assertThat(Files.toString(litsDifferencesFile, StandardCharsets.UTF_8)).isEmpty();
+  }
+
+  @Test
+  @Ignore
+  public void hbase_protocol() throws Exception {
+    File litsDifferencesFile = FileLocation.of("target/differences").getFile();
+    File pomFile = FileLocation.of("../sources/hbase/hbase-protocol/pom.xml").getFile();
+    MavenBuild mavenBuild = MavenBuild.create().setPom(pomFile).setCleanPackageSonarGoals().addArgument("-DskipTests")
+        .setProfile("rules")
+        .setProperty("sonar.cpd.skip", "true")
+        .setProperty("sonar.skipPackageDesign", "true")
+        .setProperty("sonar.analysis.mode", "preview")
+        .setProperty("sonar.issuesReport.html.enable", "true")
+        .setProperty("dump.old", FileLocation.of("src/test/resources/hbase").getFile().getAbsolutePath())
+        .setProperty("dump.new", FileLocation.of("target/actual/hbase").getFile().getAbsolutePath())
+        .setProperty("lits.differences", litsDifferencesFile.getAbsolutePath());
+    orchestrator.executeBuild(mavenBuild);
+    assertThat(Files.toString(litsDifferencesFile, StandardCharsets.UTF_8)).isEmpty();
+  }
 
+  @Test
+  public void fluent_http() throws Exception {
+    File litsDifferencesFile = FileLocation.of("target/differences").getFile();
+    File pomFile = FileLocation.of("../sources/fluent-http/pom.xml").getFile();
+    MavenBuild mavenBuild = MavenBuild.create().setPom(pomFile).setCleanPackageSonarGoals().addArgument("-DskipTests")
+        .setProfile("rules")
+        .setProperty("sonar.cpd.skip", "true")
+        .setProperty("sonar.skipPackageDesign", "true")
+        .setProperty("sonar.analysis.mode", "preview")
+        .setProperty("sonar.issuesReport.html.enable", "true")
+        .setProperty("dump.old", FileLocation.of("src/test/resources/fluent-http").getFile().getAbsolutePath())
+        .setProperty("dump.new", FileLocation.of("target/actual/fluent-http").getFile().getAbsolutePath())
+        .setProperty("lits.differences", litsDifferencesFile.getAbsolutePath());
+    orchestrator.executeBuild(mavenBuild);
```

Comment:
The construction of the {{mavenBuild}} object is mostly the same for the 4 new projects. It should be factorized.